addons/event_sale/models/sale_order.py

```diff
@@ -173,9 +167,16 @@ def _get_display_price(self):
                 pricelist=self.order_id.pricelist_id.id,
                 uom=self.product_uom.id
             )
-            if self.order_id.pricelist_id.discount_policy == 'with_discount':
-                return event_ticket.price_reduce
+            company = event_ticket.company_id or self.env.company
+            currency = company.currency_id
+            product = event_ticket.product_id
+            pricelist = product.product_tmpl_id._get_contextual_pricelist()
```

Comment:
Also here, there is no need to take the contextual pricelist, we are computing the price for a given order line, so the pricelist will always be the order one, you can keep that as well.

Reply:
Ok I changed it. I think I was using it because you said it was better. But since I'm doing it in _display_price and not in _price_reduce anymore I see why.